src/keys/utils.ts

```diff
@@ -538,6 +597,86 @@ async function decryptWithKey(
   return data.buffer.slice(data.byteOffset, data.byteOffset + data.byteLength);
 }
 
+async function writeRootKeyPair(
+  keyPair: KeyPair,
+  password: string,
+  rootPubPath: string,
+  rootKeyPath: string,
+  fs?: FileSystem,
+): Promise<void> {
+  const fs_ = fs ?? require('fs');
+  const keyPairPem = keyPairToPemEncrypted(keyPair, password);
+  // Logger.info(`Writing ${rootPubPath} and ${rootKeyPath}`);
+  try {
+    await Promise.all([
+      fs_.promises.writeFile(`${rootPubPath}.tmp`, keyPairPem.publicKey),
+      fs_.promises.writeFile(`${rootKeyPath}.tmp`, keyPairPem.privateKey),
+    ]);
+    await Promise.all([
+      fs_.promises.rename(`${rootPubPath}.tmp`, rootPubPath),
+      fs_.promises.rename(`${rootKeyPath}.tmp`, rootKeyPath),
+    ]);
+  } catch (e) {
+    throw new keysErrors.ErrorRootKeysWrite(e.message, {
+      errno: e.errno,
+      syscall: e.syscall,
+      code: e.code,
+      path: e.path,
+    });
+  }
+}
+
+async function recoverKeyManager(
```

Comment:
This I think should also be part of `KeyManager`. It's not a "reusable" utility.

Reply:
Fixed.